core/src/main/scala/org/bitcoins/core/protocol/script/ScriptWitness.scala

```diff
@@ -180,9 +180,7 @@ object ScriptWitness extends Factory[ScriptWitness] {
     //TODO: eventually only compressed public keys will be allowed in v0 scripts
     //https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#restrictions-on-public-key-type
     val isPubKey = {
-      stack.nonEmpty &&
-      ((stack.head.size == 33 && (stack.head.head == 0x02 || stack.head.head == 0x03))
-        || (stack.head.size == 65 && stack.head.head == 0x04))
+      stack.nonEmpty && (stack.head.size == 33 && (stack.head.head == 0x02 || stack.head.head == 0x03))
     }
     if (stack.isEmpty) {
       EmptyScriptWitness
```

Comment:
I think we should parse the version, then give indiviudal parsing functions for each version. So we can have `isPubKey` for v0 be different than v1

Reply:
We don't have the version i believe? Thats in the output. This is the witness stack

Reply:
Oh I see